### ISSUE TITLE
Ray fails on Windows with torchaudio (simple 3 line repro)

### ISSUE THREAD

**Issue by trzy** [ray-project/ray#42690]
### What happened + What you expected to happen

ray.init() failing in Windows on a CUDA machine. Tried installing barebones CPU-only torchaudio (`pip install torchaudio`) but still fails. Repro with this:

```
import torchaudio
import ray
ray.init()
```

And then it crashes:

```
(ai) C:\projects\UntitledAI>python
Python 3.11.7 | packaged by Anaconda, Inc. | (main, Dec 15 2023, 18:05:47) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> import torchaudio
>>> ray.init()
2024-01-24 17:40:19,380 INFO worker.py:1715 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265
[symbolize_win32.inc : 53] RAW: SymInitialize() failed: 87
Fatal Python error: Aborted

Stack (most recent call first):
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\worker.py", line 2342 in connect
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\worker.py", line 1726 in init
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\client_mode_hook.py", line 103 in wrapper
  File "<stdin>", line 1 in <module>

Extension modules: msgpack._cmsgpack, google._upb._message, psutil._psutil_windows, setproctitle, yaml._yaml, charset_normalizer.md, ray._raylet, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, torch._C, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, _cffi_backend, multidict._multidict, yarl._quoting_c, aiohttp._helpers, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket, frozenlist._frozenlist, grpc._cython.cygrpc (total: 36)
```

Seeing this in a larger project, too, but this seems to be the minimum repro.

### Versions / Dependencies

Python 3.11.7
Windows 10

Packages installed:
aiosignal==1.3.1
attrs==23.2.0
certifi==2023.11.17
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
filelock==3.13.1
frozenlist==1.4.1
fsspec==2023.12.2
idna==3.6
Jinja2==3.1.3
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
MarkupSafe==2.1.4
mpmath==1.3.0
msgpack==1.0.7
networkx==3.2.1
packaging==23.2
protobuf==4.25.2
PyYAML==6.0.1
ray==2.9.1
referencing==0.32.1
requests==2.31.0
rpds-py==0.17.1
sympy==1.12
torch==2.1.2
torchaudio==2.1.2
typing_extensions==4.9.0
urllib3==2.1.0

### Reproduction script

import torchaudio
import ray
ray.init()

### Issue Severity

High: It blocks me from completing my task.

**Comment by anyscalesam** [comment]
@mattip can you please attempt repro and root cause this?

**Comment by mattip** [comment]
I can reproduce this, from a simple installation of `ray 2.9.1` and `torchaudio==2.1.2`. 

The [error 89](https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-) rasied by `SymInitialize` (a windows-specific function to enable easier debugging) is `The system cannot start another process at this time.`.

Maybe a duplicate of #29037, which was closed without solution since it did not include a reproducer.  I see the symbol in `_raylet.pyd`, `raylet.exe`, and `gcs_server.exe`, as well as [in `pytorch`](https://github.com/peterjc123/pytorch/blob/77a7dc3f6a486f5d3f366161dcc240b8c2663883/c10/util/Backtrace.cpp#L162)  and `rpds-py` (a wrapper for the rust `rpds` crate for persistent data). In the documentation for [`SymInitialize`](https://learn.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-syminitialize) it states 
> A process that calls SymInitialize should not call it again unless it calls [SymCleanup](https://learn.microsoft.com/en-us/windows/desktop/api/dbghelp/nf-dbghelp-symcleanup) first. ... All DbgHelp functions, such as this one, are single threaded. Therefore, calls from more than one thread to this function will likely result in unexpected behavior or memory corruption. To avoid this, call SymInitialize only when your process starts and [SymCleanup](https://learn.microsoft.com/en-us/windows/desktop/api/dbghelp/nf-dbghelp-symcleanup) only when your process ends.

I think the call is coming from [absl](https://github.com/abseil/abseil-cpp/blob/797501d12ea767dabdc8d36674e083869e62ee7d/absl/debugging/symbolize_win32.inc#L48), the filename matches. The function `InitializeSymbolizer` is used in [`src/ray/logging.cc`](https://github.com/ray-project/ray/blob/c6ab40374eb6bdb2bfd2984d039a962946cdc318/src/ray/util/logging.cc#L386). This was added in #19423 by @mwtian with the comment
> Also, try to initialize symbolizer in GCS, Raylet and core worker. This is a no-op on MacOS and some Linux environments (e.g. Ray on Ubuntu 20.04 already produces symbolized stack traces), but should make Ray more likely to have symbolized stack traces on other platforms.

It would be nice to not call `absl::InitializeSymbolizer` on Windows if `SymInitialize` has already been called (I don't know how to detect this, and the windows documentation is not helpful) or if the current thread is not the main one. Another alternative, since it seems the call to `absl::InitializeSymbolizer` is "nice to have" and not a hard requirement, is to not call it at all on Windows.
